kong/plugins/key-auth/handler.lua

```diff
@@ -132,8 +132,11 @@ function KeyAuthHandler:access(conf)
   if not ok then
     if conf.anonymous ~= "" then
       -- get anonymous user
-      local consumer = cache.get_or_set(cache.consumer_key(conf.anonymous),
-                       nil, load_consumer, conf.anonymous, true)
+      local consumer, err = cache.get_or_set(cache.consumer_key(conf.anonymous),
+                            nil, load_consumer, conf.anonymous, true)
+      if err then
+        responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
+      end
```

Comment:
> Should we return HTTP 403 here if no Consumer was retrieved from the DB?

This case is not a 403, but a 500. A consumer was configured as "anonymous" but with a bad uuid (was never there), or with a consumer that was later deleted.

Hence a configuration error, and hence a 500 imo. This is also the quicker route to the maintainers that something is wrong with the configuration

> Are we sure no other plugins is guilty of the behavior I described above? aka always assuming we'll retrieve an entity from `cache.get_or_set()`?

All callbacks properly return the error, that's what I checked. But then still, this behaviour is a contract that the caller itself created between the callback he provided and his result handling code. It is not a responsibility of the caching mechanism to handle this.

Anyway, this is probably going away with an lru cache anyway, the way we discussed it.